custom_components/tuya_local/devices/idra_skiv_airconditioner.yaml

```diff
@@ -0,0 +1,347 @@
+            value: "16h"
+          - dps_val: "17"
+            value: "17h"
+          - dps_val: "18"
+            value: "18h"
+          - dps_val: "19"
+            value: "19h"
+          - dps_val: "20"
+            value: "20h"
+          - dps_val: "21"
+            value: "21h"
+          - dps_val: "22"
+            value: "22h"
+          - dps_val: "23"
+            value: "23h"
+          - dps_val: "24"
+            value: "24h"
+  - entity: sensor
+    class: duration
+    translation_key: time_remaining
+    category: diagnostic
+    dps:
+      - id: 22
+        name: sensor
+        type: integer
+        unit: min
+  - entity: sensor
+    name: Usage counter
+    category: diagnostic
+    icon: "mdi:counter"
+    dps:
+      - id: 101
+        name: sensor
+        type: integer
+  - entity: sensor
+    class: duration
+    name: Total time
+    category: diagnostic
+    icon: "mdi:timer"
+    dps:
+      - id: 102
+        name: sensor
+        type: integer
+        unit: h
+  - entity: sensor
+    name: Electricity
+    category: diagnostic
+    icon: "mdi:check-circle"
+    dps:
+      - id: 103
+        name: sensor
+        type: integer
+        unit: kWh
+        mapping:
+          - scale: 10
+      - id: 104
+        type: integer
+        name: report_number
+        optional: true
+  - entity: select
+    translation_key: temperature_unit
+    category: config
+    dps:
+      - id: 105
+        type: string
+        optional: true
+        name: option
+        mapping:
+          - dps_val: c
+            value: celsius
+          - dps_val: f
+            value: fahrenheit
+  - entity: select
+    name: Vertical
+    category: config
+    icon: "mdi:arrow-up-down"
+    dps:
+      - id: 107
+        type: string
+        name: option
+        mapping:
+          - dps_val: "off"
+            value: "Off"
+          - dps_val: "15"
+            value: "Swing"
+          - dps_val: "1"
+            value: "1"
+          - dps_val: "2"
+            value: "2"
+          - dps_val: "3"
+            value: "3"
+          - dps_val: "4"
+            value: "4"
+          - dps_val: "5"
+            value: "5"
+  - entity: switch
+    translation_key: sleep
+    category: config
+    dps:
+      - id: 109
+        name: switch
+        type: boolean
+  - entity: switch
+    name: Plasma Filter
+    icon: "mdi:air-filter"
+    dps:
+      - id: 110
+        name: switch
+        type: boolean
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    icon: "mdi:check-circle-outline"
+    dps:
+      - id: 20
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: true
+            constraint: fault_code2
+            conditions:
+              - dps_val: 0
+                value: false
+          - value: true
+      - id: 20
+        type: bitfield
+        name: fault_code
+      - id: 20
+        type: bitfield
+        name: Fault Description
```

Comment:
Attribute names should not contain capital letters or spaces. Normally "description" is used for this. Icon normally is not overridden when a class or translation_key defines an icon already, as this can interfere with user theming.